Terminal.Gui/ConsoleDrivers/CursesDriver/UnixMainLoop.cs

```diff
@@ -206,7 +193,11 @@ bool CkeckTimeout (bool wait, ref int pollTimeout)
 			if (!wait)
 				pollTimeout = 0;
 
-			return false;
+			int ic;
+			lock (mainLoop.idleHandlers)
```

Comment:
Please put `{` and `}` around this lock. We like to use this style for all nested statements as it makes intent super clear and  reduces the chance of bugs.